### PR TITLE
Feature/tao 8584/deprecate php serialization service call

### PR DESCRIPTION
**Pull request by SashaTAO**
JIRA: [TAO-8584](https://oat-sa.atlassian.net/browse/TAO-8584)

Replace PHP serialization/unserialization of ServiceCall with JSON implementation.
Render postInstall command report in cli.

### CODE REVIEW

File: test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
+
```
Comment (zagovorichev) [inline test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: I don't think that this is a good name for the test-file, can you use shorter?

Comment (zagovorichev) [inline test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: copyright?

File: models/classes/service/class.ServiceCall.php [hunk 2]
```diff
@@ -169,21 +170,32 @@ public static function fromResource(core_kernel_classes_Resource $resource) {
...
-	    return serialize($this);
+	    return json_encode($this);
 	}
 	
 	/**
 	 * Unserialize the string to a serivceCall object
 	 * 
 	 * @param string $string
 	 * @return tao_models_classes_service_ServiceCall
+     * @throws UserErrorException
```
Comment (zagovorichev) [inline models/classes/service/class.ServiceCall.php]: deprecated?

File: install/class.Setup.php [hunk 3]
```diff
@@ -318,7 +318,9 @@ public function __invoke($params)
                         $object->setServiceLocator($serviceManager);
                     }
                     $params = (isset($script['params']) && is_array($script['params'])) ? $script['params'] : [];
-                    call_user_func($object, $params);
+                    $report = call_user_func($object, $params);
+
+                    $this->logInfo(helpers_Report::renderToCommandline($report));
```
Comment (zagovorichev) [inline install/class.Setup.php]: could you check that $report is an instance of the Report?

File: models/classes/service/class.ServiceCall.php (hunk 2 above)
Comment (zagovorichev) [inline models/classes/service/class.ServiceCall.php]: InvalidArgumentException?

File: scripts/update/Updater.php [hunk 4]
```diff
@@ -1269,6 +1269,6 @@ public function update($initialVersion) {
             $this->setVersion('39.6.0');
          }
 
-        $this->skip('39.6.0', '39.6.1');
+        $this->skip('39.6.0', '39.7.0');
```
Comment (zagovorichev) [inline scripts/update/Updater.php]: these changes break down the backward compatibility

File: test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+use InvalidArgumentException;
+use tao_models_classes_service_Parameter;
+use tao_models_classes_service_ServiceCall;
+use tao_models_classes_service_VariableParameter;
+use oat\generis\test\MockObject;
+use oat\generis\test\TestCase;
+
+class tao_models_classes_service_ServiceCallTest extends TestCase
+{
+    const SERVICE_DEFINITION = 'DEFINITION';
```
Comment (zagovorichev) [inline test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: identifier, or service_definition?

File: test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php (hunk 1 above)
Comment (SashaTAO) [reply test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: Fixed

Comment (SashaTAO) [reply test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: Fixed

File: models/classes/service/class.ServiceCall.php (hunk 2 above)
Comment (SashaTAO) [reply models/classes/service/class.ServiceCall.php]: Fixed

File: install/class.Setup.php (hunk 3 above)
Comment (SashaTAO) [reply install/class.Setup.php]: Done

File: models/classes/service/class.ServiceCall.php (hunk 2 above)
Comment (SashaTAO) [reply models/classes/service/class.ServiceCall.php]: Done

File: test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php (hunk 5 above)
Comment (SashaTAO) [reply test/unit/models/classes/service/tao_models_classes_service_ServiceCallTest.php]: Didn't clearly understand your question. SERVICE_DEFINITION here is just a fake service definition URI required in constructor. In terms of test string value doesn't really matter, but updated it to make it more clear.

File: scripts/update/Updater.php (hunk 4 above)
Comment (SashaTAO) [reply scripts/update/Updater.php]: Updated version

**Pull request merged**